### PR TITLE
gh-315 Create EditSafeName screen

### PR DESCRIPTION
**Pull request by MouazAlzahabi**

### CODE REVIEW

File: Multisig/UI/App/UIKit/Settings/SafeSettingsViewController/SafeSettingsViewController.swift [hunk 1]
```diff
@@ -256,8 +256,8 @@ class SafeSettingsViewController: LoadableViewController, UITableViewDelegate, U
         let item = sections[indexPath.section].items[indexPath.row]
         switch item {
         case Section.Name.name(_):
-            let vc = ViewControllerFactory.editSafeNameController(address: safe.address, name: safe.name, presenter: self)
-            present(vc, animated: true, completion: nil)
+            let vc = ViewControllerFactory.editSafeNameController(address: safe.addressValue, name: safe.name, presenter: self)
+            show(vc, sender: self)
```
Comment (DmitryBespalov) [inline Multisig/UI/App/UIKit/Settings/SafeSettingsViewController/SafeSettingsViewController.swift]: why don't we just create the edit safe name view controller directly here? The previous "viewcontrollerfactory" was there as a way to put all swiftui-related controller creation in one place

File: Multisig/UI/App/UIKit/Safe Management/Edit Safe/EditSafeNameViewController.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    private var saveButton: UIBarButtonItem!
+    @IBOutlet private weak var textField: GNOTextField!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        assert(address != nil, "Developer error: expect to have an address")
+
+        navigationItem.title = "Edit Safe Name"
+
+        saveButton = UIBarButtonItem(title: "save", style: .done, target: self, action: #selector(didTapSaveButton))
```
Comment (DmitryBespalov) [inline Multisig/UI/App/UIKit/Safe Management/Edit Safe/EditSafeNameViewController.swift]: is the title correct? here it starts with small letter

File: Multisig/UI/App/UIKit/Settings/SafeSettingsViewController/SafeSettingsViewController.swift (hunk 1 above)
Comment (MouazAlzahabi) [reply Multisig/UI/App/UIKit/Settings/SafeSettingsViewController/SafeSettingsViewController.swift]: I wanted to keep the consistency of our code base, but I will change it

File: Multisig/UI/App/UIKit/App/ViewControllerFactory.swift [hunk 3]
```diff
@@ -154,6 +140,10 @@ enum ViewControllerFactory {
     func closeModal()
 }
 
+@objc protocol PresentingViewController {
```
Comment (DmitryBespalov) [inline Multisig/UI/App/UIKit/App/ViewControllerFactory.swift]: we don't need this protocol anymore, right?

**Review by DmitryBespalov — APPROVED** [review]

**Review by DmitryBespalov — APPROVED** [review]

**Pull request merged**